src/main/java/thaumicenergistics/common/network/IAspectSlotPart.java

```diff
@@ -10,5 +11,10 @@
  *
  */
 public interface IAspectSlotPart {
+    int[] getAvailableAspectSlots();
+
+    @Nullable
+    Aspect getAspect(int index);
```

Comment:
I think it would not harm to add `default` implementations for these methods

Reply:
I could, but I’m having a hard time thinking of what sensible defaults would be. Returning null would be a safe default for getAspect, and the only safe default for getAvailableAspectSlots would be an empty array.

If you look at the 3 existing implementations of these methods in the PR, they share nothing in common with each other due to being very item-specific. So, for other code to reliably make use of this new API, it would really need each class to ship its own implementation that makes sense for that class.

For my specific use-case it’s a non-issue since the classes I care about have such an implementation. But the “safe but useless” defaults I proposed above would lay a trap for any future usage in other classes. I think the main downside of not shipping a default is other mods implementing this interface as well suddenly failing to build until they provide such implementations (so, compile time error in mods exposing the API vs. silently doing the wrong thing at runtime for mods using the API). I’ll do some searching to see which (if any) mods would be impacted by this.

If you’d still like to see defaults despite those issues, or have ideas for better defaults than what I gave, I’m all ears.

Reply:
I think empty array is ok, I don't see it as a trap, on the other hand I'm not sure if any other mods implement that interface